lib/common/xml.c

```diff
@@ -1446,37 +1446,56 @@ crm_xml_escape(const char *text)
     return copy;
 }
 
-static inline void
-dump_xml_attr(xmlAttrPtr attr, int options, char **buffer, int *offset, int *max)
+/*!
+ * \internal
+ * \brief Append an XML attribute to a buffer
+ *
+ * \param[in]     attr     Attribute to append
+ * \param[in]     options  Group of \p xml_log_options flags
+ * \param[in,out] buffer   Where to append the content, if not \p NULL
+ *
+ * \return The modified \p buffer, or a newly allocated one initialized
+ *         with the XML attribute if \p buffer is \p NULL
+ *
+ * \note The caller is responsible for freeing the string using
+ *       \p g_string_free().
+ */
+static inline GString *
+dump_xml_attr(const xmlAttr *attr, int options, GString *buffer)
 {
     char *p_value = NULL;
     const char *p_name = NULL;
     xml_private_t *p = NULL;
 
     CRM_ASSERT(buffer != NULL);
     if (attr == NULL || attr->children == NULL) {
-        return;
+        return buffer;
     }
 
     p = attr->_private;
     if (p && pcmk_is_set(p->flags, pcmk__xf_deleted)) {
-        return;
+        return buffer;
+    }
+
+    // Probably shouldn't happen
+    if (buffer == NULL) {
+        buffer = g_string_sized_new(128);
     }
 
-    p_name = (const char *)attr->name;
+    p_name = (const gchar *) attr->name;
```

Comment:
I need to make `p_name` is a `const gchar *` variable if I'm going to do this. Probably better to just cast it in the `g_string_append_printf()` call.

Reply:
Most of the casts aren't really necessary, just a safeguard in case static analysis complains later.